### ISSUE TITLE
Vertical scroll bar stops halfway down

### ISSUE THREAD

**Issue by mjross** [Komodo/KomodoEdit#3373]
### Summary
In a file long enough to generate a vertical scroll bar, scroll bar handle (a.k.a. "elevator") does not go all the way down.

### Steps to Reproduce
Help > Troubleshooting > Restart in Safe-Mode. In a file long enough to generate a vertical scroll bar, PageDown to the bottom of the file.

### Expected Results
The scroll bar handle should reach the bottom of the scroll bar region.

### Actual Results
It stops about halfway down.

### Platform Information
Komodo IDE 11.0.1 build 90797 on Windows 10.0.16299.19 Pro

### Attachments
The screenshot shows a file containing 37 lines total, i.e., the view is scrolled all the way to the bottom of the file.
![clipboard01](https://user-images.githubusercontent.com/829492/32983506-42ab8468-cc5b-11e7-8f28-dd7141d5f276.jpg)

**Comment by mitchell-as** [comment]
I am not able to reproduce this on Linux. @cgchoffman Can you reproduce this on Windows?

**Comment by cgchoffman** [comment]
Are you sure you're in safe mode?  Word wrap shouldn't be on if you're in safe mode, unless of course you turned it on manually.

I can't reproduce this with or without wordwrap.

**Comment by mjross** [comment]
Yes, I'm in safe mode. I turned on word wrap to generate the vertical scroll bar and because that's the mode in which I always edit files. I should've mentioned that in my bug report.

I tried it again, but without turning on word wrap, and got the same behavior.

![clipboard01](https://user-images.githubusercontent.com/829492/33036016-cb5c2b6c-cdf2-11e7-8e6b-7cb77da4ab2a.jpg)

**Comment by cgchoffman** [comment]
That scroll bar looks like it's at the bottom to me.

**Comment by mjross** [comment]
You are correct. Sorry about that. As I was paging down in the file, the scrollbar handle was not going all the way to the bottom of the scrollbar region. Perhaps there was a delay between the time that I observed that and when I pressed the key combination to create a screenshot of just that application; maybe during that second or two, the handle completed its journey to the bottom. I should've been paying more attention to the screen just prior to taking the screenshot. (Komodo IDE 11 often seems more sluggish than version 10, especially page refreshes.)

When I enable word wrap, then the problem appears. With my edit cursor at the bottom of the file, paging down further does nothing (of course), but the scrollbar handle is about halfway down, as seen in the first screenshot above. So I have amended the original bug report to indicate that.

**Comment by cgchoffman** [comment]
Hmm, i doubt this will work but could you try attaching the file you're seeing this in?  I still can't reproduce this with Word Wrap enabled.

**Comment by mjross** [comment]
Certainly.

[Text-1.txt](https://github.com/Komodo/KomodoEdit/files/1488935/Text-1.txt)

[pystderr.log.txt](https://github.com/Komodo/KomodoEdit/files/1488936/pystderr.log.txt)

**Comment by cgchoffman** [comment]
No luck with the file and nothing helpful in the logs.  Any ideas from the scintilla side @mitchell-as ?

**Comment by mitchell-as** [comment]
No, Scintilla calculates its scroll position correctly. The only potential problem I can envision is if we're using our own scroll bar (not Scintilla's) and calculating its position manually.

**Comment by Naatan** [comment]
Cannot repro either. @mjross are you certain you didn't miss a step?

**Comment by mjross** [comment]
I'm confidant no step was missed. Perhaps a screencast would help?

The Xbox app in Windows 10 has a video recorder tool intended to let you capture in game videos. [It will work with just about any app](https://windows.gadgethacks.com/how-to/use-hidden-screen-recorder-tool-windows-10-0163266/). It does not capture the app's drop-down menus (sorry about that, but I'm hesitant to begin installing and testing other video recorder tools because if one of them hoses my system, I'm SOL, as there's little tech support/sales here in Nicaragua). But it does get the window contents and the mouse pointer. I hope that's enough to help. 

I recorded all the steps, but Github won't let me attach an MP4 file, so here it is: www.ross.ws/issue_3373.mp4

In making the video, I discovered that the issue does not happen when the handle is pulled down with the mouse pointer.

**Comment by Naatan** [comment]
Reproduced! Not sure if this was key, but I went to the last line first - then when turning on word wrap changed the caret to the first line and then press pg down.

**Comment by mjross** [comment]
Thank you, kind sir! I was getting worried my laptop (or brain...) might be
infected with ghosts. ;)
